.llvm-ci.yml

```diff
@@ -0,0 +1,88 @@
+resources:
+  repositories:
+  - repository: llvm
+    type: github
+    endpoint: LLVMServiceConnection
+    name: llvm/llvm-project
+
+jobs:
+- job:
+  displayName: LLVM Linux Build
+  pool: verona-llvm-build-linux
+  timeoutInMinutes: 120
+  strategy:
+    matrix:
+      Clang Release:
+        CC: clang
+        CXX: clang++
+        CXXFLAGS: -stdlib=libstdc++
+        BuildType: Release
+        BuildName: release
+        Sanitizer:
+      Clang Debug+ASAN:
+        CC: clang
+        CXX: clang++
+        CXXFLAGS: -stdlib=libstdc++
+        BuildType: Debug
+        BuildName: debug
+        Sanitizer: Address
+  steps:
+  - checkout: llvm
+
+  - script: |
+      set -eo pipefail
+      git checkout master
+      git fetch
+      git pull
+      if git branch | grep -q "^  verona$"; then
+        git branch -D verona
+      fi
+      git checkout -b verona $(GOOD_HASH)
+    displayName: 'Move tree to known good LLVM commit for Verona'
+
+  - script: |
+      set -eo pipefail
+      sudo apt-get update
+      sudo apt-get install -y clang ninja-build cmake lld
+    displayName: 'Install Build Dependencies'
+
+  - script: |
+      set -eo pipefail
+      rm -rf /tmp/VeronaLLVM*
+    displayName: 'Cleanup'
+
+  - task: CMake@1
+    displayName: 'CMake'
+    inputs:
+      cmakeArgs: |
+        ../llvm -GNinja -DCMAKE_BUILD_TYPE=$(BuildType) -DCMAKE_C_COMPILER=$(CC) -DCMAKE_CXX_COMPILER=$(CXX) -DCMAKE_CXX_FLAGS=$(CXXFLAGS) -DLLVM_USE_SANITIZER=$(Sanitizer) -DLLVM_ENABLE_PROJECTS=mlir -DLLVM_BUILD_EXAMPLES=ON -DLLVM_TARGETS_TO_BUILD="X86" -DCMAKE_BUILD_TYPE=Debug -DLLVM_ENABLE_ASSERTIONS=ON -DLLVM_ENABLE_LLD=ON
+
+  - script: |
+      set -eo pipefail
+      ninja
+      VERSION=$(./bin/mlir-tblgen --version | grep -o "[0-9]\+\.[0-9]\+\.[0-9]\+" | cut -d "." -f 1,3)
+      echo "##vso[task.setvariable variable=VERSION]$VERSION"
+    workingDirectory: build
```

Comment:
I prefer build metadata than script logic, because this way the behaviour is the same on all platforms without having to write platform-independent code (DevOps does that for you).

Same argument for the CMake task.

Reply:
If you want more cross-platformness then you should use `bash:` instead of `script:` which will use Git bash on Windows.

Reply:
Right now I want to get the build running. :)
Given that this is derived from them main build, I prefer to repeat the same pattern, so it's easier to maintain. If that's not optimal, we can change the style on all scripts on a different commit.